R/sharepoint_folder.R

```diff
@@ -93,6 +93,36 @@ sharepoint_folder <- R6::R6Class(
       rbind(folders[v], files[v])
     },
 
+    #' @description Delete a folder. Be extremely careful as you could
+    #' use this to delete an entire sharepoint. Deleted files are sent
+    #' to the recycle bin, so can be restored with relative ease, but
+    #' it will still be alarming. There is a mechanism to prevent
+    #' accidental deletion by declaring a file that exists within the
+    #' folder.
+    #'
+    #' @param path The path to delete. Use \code{NULL} to delete the current
+    #'   folder.
+    #'
+    #' @param check A file (not folder) that exists directly within
+    #'   \code{path}, used as a method to verify that you really do want
+    #'   to delete this folder (to prevent things like accidental deletion
+    #'   of the entire sharepoint, for example).
+    delete = function(path, check) {
+      if (!(check %in% self$files(path)$name)) {
```

Comment:
I wonder if something like if you want to delete a folder it creates a confirm dialog where you have to write the name of the folder to confirm the delete. Kind of like github does when deleting a repo. Though I think this pattern probably fine, good that it goes into a recycle bin. Also maybe worth thinking if we can add a function which will reverse the delete action? Could move files back from the recycle bin - though maybe after an accidental delete it is easier to recover state through UI.

Reply:
I do need to do it in somewhat unsupervised code; use case is here: https://github.com/mrc-ide/roundtable/blob/prototype/R/rtm.R#L215-L216